src/hotspot/cpu/aarch64/vm_version_aarch64.cpp

```diff
@@ -408,6 +408,25 @@ void VM_Version::initialize() {
     UsePopCountInstruction = true;
   }
 
+#ifdef __ARM_FEATURE_PAC_DEFAULT
+  if (_features & CPU_PACA) {
+    if (FLAG_IS_DEFAULT(UseROPProtection)) {
+      FLAG_SET_DEFAULT(UseROPProtection, true);
+    }
+  } else {
+    if (UseROPProtection) {
+      warning("UseROPProtection specified, but not supported on this CPU.");
+      FLAG_SET_DEFAULT(UseROPProtection, false);
```

Comment:
```suggestion
      FLAG_SET_DEFAULT(UseROPProtection, true);
```
Given that the instructions used are in NOP space, this won't do any harm, and it will allow developers without PAC-enabled systems to see what code PAC would generate.

Reply:
Ok, I think that's fine. How about on a non pac system allowing it for development only ?

Reply:
Maybe. Mind you, a lot of the time I'm looking at the output from production systems.
From a rather philosophical point of view, I assume that if the user of a computer asks for something that isn't going to break anything or confuse anyone, we should honour their request.

Reply:
Was this ever resolved?

Reply:
Sort of. That code has changed quite a bit - UseROPProtection now is a string field not a bool.
"none" or not set - pac disabled.
"pac-ret" - pac always enabled
"standard" - pac enabled if the cpu+os support it.

Also, the pac instructions used aren't all in the NOP space. So, it will crash on a non-pac machine.  It might be possible to change it so it does only use nop space instructions, but I don't think it'll be optimal (need to double check).

Reply:
OK, cool.